src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -368,6 +368,7 @@ private static ServiceInterface createServicesFromPlan(String configName, String
       return null;
     }
 
+    Set<String> autoStartedPeers = new HashSet<>();

```

Comment:
why is this defined locally if it's also defined on the base Service class?  that's confusing.

Reply:
because when Runtime is requested to start a composite service that has peers, the current contract is that it starts the peers first "before" the composite service is started.   You can't use si.addAutoStartedPeer(peerName) "before" si is created